### PR TITLE
20191119 stv notes

### PR DESCRIPTION
**Pull request by stvoutsin**

### CODE REVIEW

File: notes/stv/20191209-openstack-deployment.txt [hunk 1]
```diff
@@ -0,0 +1,919 @@
...
+#-------------------------------------------------------------------------
+
+~/.ssh/config
+..
+
+
+Host stv-dev-gateway
+    User centos
+    IdentityFile ~/.ssh/stv-master
+    Hostname ${stv-dev-gateway-ip:?}
```
Comment (Zarquan) [inline notes/stv/20191209-openstack-deployment.txt]: This bash variable substitution wouldn't work in ssh so how was it resolved ?

File: notes/stv/20191209-openstack-deployment.txt [hunk 2]
```diff
@@ -0,0 +1,919 @@
...
+stv-dev-worker-2-ip=
+stv-dev-worker-3-ip=
+stv-dev-master-ip=
+stv-dev-zeppelin-ip=
+
+
+
+
+#-------------------------------------------------------------------------
+# Setup .ssh config locally
```
Comment (Zarquan) [inline notes/stv/20191209-openstack-deployment.txt]: locally is on laptop ?

File: notes/stv/20191209-openstack-deployment.txt [hunk 3]
```diff
@@ -0,0 +1,919 @@
...
+
+
+#-------------------------------------------------------------------------
+## Step 5: Install HDFS Cluster
+#-------------------------------------------------------------------------
+
+## Fetch Hadoop Binaries on each node & install java
+
+
+## Run on all nodes
```
Comment (Zarquan) [inline notes/stv/20191209-openstack-deployment.txt]: What location is the tar.gz unpacked ?

File: notes/stv/20191209-openstack-deployment.txt (hunk 2 above)
Comment (stvoutsin) [reply notes/stv/20191209-openstack-deployment.txt]: Yes, locally on laptop, notes updated to clarify this.

File: notes/stv/20191209-openstack-deployment.txt (hunk 1 above)
Comment (stvoutsin) [reply notes/stv/20191209-openstack-deployment.txt]: The placeholders were added before committing the notes, mainly to hide the public IPs in github. I've updated the notes to change dashes to underscores.

File: notes/stv/20191209-openstack-deployment.txt (hunk 3 above)
Comment (stvoutsin) [reply notes/stv/20191209-openstack-deployment.txt]: tar.gz was unpacked under the fedora user's home directory  (/home/fedora). Notes updated to make this clearer.

**Comment by stvoutsin** [comment]
I've removed the binary ascii 2 parquet file, and changed the example notebook to a pretty-printed JSON file.

File: notes/stv/20191209-openstack-deployment.txt (hunk 1 above)
Comment (Zarquan) [reply notes/stv/20191209-openstack-deployment.txt]: The ${something:?} syntax still looks like a bash variable substitution.
If you want to just show a manual substitution, just {something} without the $ and :? might be clearer.
I wouldn't worry too much about revealing public IP addresses, the bad guys already know them.

Comment (Zarquan) [reply notes/stv/20191209-openstack-deployment.txt]: Ah - ok, my bad. I was looking at the four line fragment above, and didn't realise what the 'Outdated' label meant. It meant the fragment was out of date and I needed to go look at the latest version of the file.

Comment (Zarquan) [reply notes/stv/20191209-openstack-deployment.txt]: Adding the `cat << EOF >> ~/.ssh/config` makes it clear how the substitution happened.

File: notes/stv/20191209-openstack-deployment.txt (hunk 2 above)
Comment (Zarquan) [reply notes/stv/20191209-openstack-deployment.txt]: Yep, thanks, clearer now.

**Pull request merged**